AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -1121,7 +1121,13 @@ - (void)notifyWithUserConfirmation:(MSUserConfirmation)userConfirmation {
      */
     [MS_USER_DEFAULTS setObject:[[NSNumber alloc] initWithBool:YES] forKey:kMSUserConfirmationKey];
   }
-
+  
+  // Check if there is no handler set and unprocessedReports are not initialized as NSMutableArray (Init occurs in correct call sequence).
+  if ([MSCrashes sharedInstance].userConfirmationHandler==nil && !self.unprocessedReports) {
+    MSLogWarning(MSCrashes.logTag, @"Incorrect implementation of notifyWithUserConfirmation: should only be called from userConfirmationHandler. See https://docs.microsoft.com/en-us/appcenter/sdk/crashes/xamarin#ask-for-the-users-consent-to-send-a-crash-log");
```

Comment:
I don't think we need to paste the explicit link to the online documentation. We don't want to get back to the code if the link to the doc change. `For more information refer to the online documentation` or something is probably enough.